base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -1401,16 +1416,52 @@ where
             .await
             .map_err(|e| TransactionServiceProtocolError::new(tx_id, e.into()))?;
         let sender_message = TransactionSenderMessage::new_single_round_message(stp.get_single_round_message()?);
+        // TODO: Save spending key in key manager
         let spend_key = PrivateKey::random(&mut OsRng);
         let rtp = ReceiverTransactionProtocol::new(
             sender_message,
             PrivateKey::random(&mut OsRng),
-            spend_key,
+            spend_key.clone(),
             &self.resources.factories,
         );
 
         let recipient_reply = rtp.get_signed_data()?.clone();
-
+        let commitment = recipient_reply.output.commitment.clone();
+        let range_proof = recipient_reply.output.proof.clone();
+        let (nonce_a, _pub_nonce_a) = PublicKey::random_keypair(&mut OsRng);
```

Comment:
```suggestion
        let nonce_a, = PrivateKey::random(&mut OsRng);
```
You only care about the secret key, so just generate a secret key here

Reply:
same with nonce_x

Reply:
nit(perf): +1 the signature sign method does the nonce -> public nonce work so we can just generate a private nonce.